src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -484,7 +484,7 @@ public void onDestroyed()
      */
     public TileEntityColonyBuilding getTileEntity()
     {
-        if (tileEntity == null && colony.getWorld().getBlockState(location).getBlock() != null)
+        if ((tileEntity == null && colony.getWorld().getBlockState(location).getBlock() != null) || tileEntity.isInvalid())
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'tileEntity' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2259)